drivers/tty/serial/serial_core.c

```diff
@@ -3404,6 +3404,38 @@ int serial_core_register_port(struct uart_driver *drv, struct uart_port *port)
 	 */
 	port->flags |= UPF_DEAD;
 
+	if (port->dev == NULL) {
+		const char *driver_name = drv->driver_name;
+		
+		if (strncmp(driver_name, "niserial987x", 12) == 0) {
+			// Max number of cRIO modules = 8
+			// Max number of uart ports per module = 4
+			const char *const platform_device_names[32] = {
```

Comment:
If you make this a 2-D array then it will be a lot easier to index into using the form:
`platform_device_names[slot_number - 1][port->line]`

Probably make it static just to make sure this doesn't get initialized at runtime.
You could also use a shorter name just to waste less space.

```
static const char * const [8][4] = {
        { "987x_1_1", "987x_1_2", "987x_1_3", "987x_1_4" },
        ...
```

Reply:
Looking at how platform_device_register_simple works, it has to point to strings that will remain valid after the function returns, so it should definitely be static.

It would make sense to move it out of the function...but it is nice that in the current location it keeps the patch at one added block, and not multiple that have to merge in different places.